platforms/software/dependency-management/src/main/java/org/gradle/internal/resource/transfer/DefaultCacheAwareExternalResourceAccessor.java

```diff
@@ -144,10 +146,17 @@ public LocallyAvailableExternalResource getResource(final ExternalResourceName l
             }
 
             // All local/cached options failed, get directly
-            return copyToCache(location, fileStore, delegate.withProgressLogging().resource(location, revalidate));
+            File cachePosition = calculateFixedCachePosition(location);
+            return copyToCache(location, fileStore, delegate.withProgressLogging().resource(location, revalidate, cachePosition));
         });
     }
 
+    @Nullable
+    private File calculateFixedCachePosition(@Nonnull ExternalResourceName location) {
+        // FIXME 2024/9/5: calculate a fixed position by location
+        return null;
+    }
+
```

Comment:
@ov7a How do I calculate a proper fixed cache position that is going to be used to store the partially downloaded file in this method ? 🤔

Any suggestion ?

Reply:
I don't have a great answer for that. I can suggest that you derive it from the location (location hash?).

FTR, once you're done, the PR will be sent to the team responsible for this code area, and they may have better suggestions on how to organize code/handle issue in general